docs/fsharp/style-guide/component-design-guidelines.md

```diff
@@ -693,11 +693,83 @@ let checkNonNull argName (arg: obj) =
     | null -> nullArg argName
     | _ -> ()
 
-let checkNonNull` argName (arg: obj) =
+let checkNonNull' argName (arg: obj) =
     if isNull arg then nullArg argName
     else ()
 ```
 
+Starting with F# 9, you can leverage the new `| null` [syntax](../language-reference/values/null-values.md#null-values-starting-with-f-9) to make the compiler indicate possible null values and where they need handling. The above code will then start producing warnings:
+
+```fsharp
+let checkNonNull argName (arg: obj) =
```

Comment:
The code does not relate to the text.
Also the text mentiones "above code" but should say "below code" ?


More importantly, the text and indeed the main motivation is to warn on possible nullness exceptions.
This code sample rather shows a "useless null check", since the code is overly defensive in handling null where it cannot be present.

Better starting example is (string|null).Length and a warning for it.

Reply:
The point here was to say that the same (above) code would start producing warnings in F# 9.
But if that's not too clear, that should be indeed rewritten.

I removed some text and examples to decrease the content load here.